synapse/rest/client/v1/room.py

```diff
@@ -230,6 +233,14 @@ async def on_POST(self, request, room_id, event_type, txn_id=None):
             "sender": requester.user.to_string(),
         }
 
+        inherit_depth = False
+        prev_events = parse_strings_from_args(request.args, "prev_event")
+        if self._msc2716_enabled:
+            if prev_events and requester.app_service:
```

Comment:
Instead of just ignoring the `?prev_event` query param if the user is not an app service, should we throw a 403 error?

Same for the existing `?ts` parameter below?

Reply:
Preferably not for `ts`, that would break stuff in my bridges and be somewhat annoying to work around. I always pass the `ts` parameter even when sending messages with double puppeting:

* if it doesn't work, then the timestamp is a bit off, no big deal.
* if the server is modified to allow timestamp massaging with double puppeting, then it does work and the bridge didn't even need to know that it's supported.

Might make sense for `prev_event` where the effects of the parameter being ignored are more serious (event goes in the completely wrong place)

Reply:
Resolving this thread as we're now using our own new batch sending endpoint. The logic in `/send` is unchanged.